src/test/java/tests/FilterLexerTests.java

```diff
@@ -37,8 +37,8 @@ public void lexer() {
             new Lexer("a,b").lex();
         } catch (ParseException ignored) {}
 
-        assertEquals(new Lexer("sort: a, b ").lex(), Arrays.asList(
-            new Token(TokenType.QUALIFIER, "sort:"),
+        assertEquals(new Lexer("sort-self: a, b ").lex(), Arrays.asList(
+            new Token(TokenType.QUALIFIER, "sort-self:"),
```

Comment:
Don't change this test, just add another. You can also cover more pathological cases, like `-aaa-bbb`, `---aaa-bbb`, `aaa-bbb-`, etc.